url78eec0e0b6_Orikuro_vic_tgz-pJ8-net_nexon_vindictus_itemcomparer_inport_StaticImportJ8/src/net/nexon/vindictus/itemcomparer/logic/ItemLogic.java

```diff
@@ -0,0 +1,622 @@
+		if (ignore.length() == 0) {
+			return;
+		}
+		String setname;
+
+		List<String> names = new ArrayList<String>();
+
+		if (!ignore.contains(",")) {
+			names.add(ignore);
+		} else {
+			String[] igar = ignore.split(",");
+			for (String x : igar) {
+				names.add(x.trim());
+			}
+		}
+
+		int count = 0;
+		List<Shoes> sho = new ArrayList<>();
+		List<Pants> pan = new ArrayList<>();
+		List<Gloves> glo = new ArrayList<>();
+		List<Armor> arm = new ArrayList<>();
+		List<Helm> hel = new ArrayList<>();
+		for (String igname : names) {
+			for (Shoes s : items.getShoes()) {
+				setname = s.getItemset().getName();
+				if (setname.toLowerCase().contains(igname)) {
+					sho.add(s);
+					count++;
+				}
+			}
+			for (Pants s : items.getPants()) {
+				setname = s.getItemset().getName();
+				if (setname.toLowerCase().contains(igname)) {
+					pan.add(s);
+					count++;
+				}
+			}
+			for (Gloves s : items.getGloves()) {
+				setname = s.getItemset().getName();
+				if (setname.toLowerCase().contains(igname)) {
+					glo.add(s);
+					count++;
+				}
+			}
+			for (Armor s : items.getArmors()) {
+				setname = s.getItemset().getName();
+				if (setname.toLowerCase().contains(igname)) {
+					arm.add(s);
+					count++;
+				}
+			}
+			for (Helm s : items.getHelms()) {
+				setname = s.getItemset().getName();
+				if (setname.toLowerCase().contains(igname)) {
+					hel.add(s);
+					count++;
+				}
+			}
+		}
+		items.getShoes().removeAll(sho);
+		items.getPants().removeAll(pan);
+		items.getHelms().removeAll(hel);
+		items.getArmors().removeAll(arm);
+		items.getGloves().removeAll(glo);
+		System.out.println(count + " Items removed due to ignore ( " + ignore
+				+ " )");
+
+	}
+
+	public void filterPrice(Items items, double price) {
+		if (price == Double.MAX_VALUE) {
+			return;
+		}
+
+		int count = 0;
+		List<Shoes> sho = new ArrayList<>();
+		List<Pants> pan = new ArrayList<>();
+		List<Gloves> glo = new ArrayList<>();
+		List<Armor> arm = new ArrayList<>();
+		List<Helm> hel = new ArrayList<>();
+
+		for (Shoes s : items.getShoes()) {
+			if (s.getTotalPrice() > price) {
+				sho.add(s);
+				count++;
+			}
+		}
+		for (Pants s : items.getPants()) {
+			if (s.getTotalPrice() > price) {
+				pan.add(s);
+				count++;
+			}
+		}
+		for (Gloves s : items.getGloves()) {
+			if (s.getTotalPrice() > price) {
+				glo.add(s);
+				count++;
+			}
+		}
+		for (Armor s : items.getArmors()) {
+			if (s.getTotalPrice() > price) {
+				arm.add(s);
+				count++;
+			}
+		}
+		for (Helm s : items.getHelms()) {
+			if (s.getTotalPrice() > price) {
+				hel.add(s);
+				count++;
+			}
+		}
+
+		items.getShoes().removeAll(sho);
+		items.getPants().removeAll(pan);
+		items.getHelms().removeAll(hel);
+		items.getArmors().removeAll(arm);
+		items.getGloves().removeAll(glo);
+		System.out.println(count + " Items removed due to price ( " + price
+				+ " )");
+	}
+
+	public void filterLevel(Items items, int min, int max) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 16. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 331-351 into a separate method.